ethereum/spec/src/testFixtures/java/tech/pegasys/teku/spec/executionengine/StubExecutionEngineChannel.java

```diff
@@ -42,19 +69,72 @@ public void addPowBlock(final PowBlock block) {
   }
 
   @Override
-  public SafeFuture<Void> forkChoiceUpdated(
+  public SafeFuture<ForkChoiceUpdatedResult> forkChoiceUpdated(
       final ForkChoiceState forkChoiceState, final Optional<PayloadAttributes> payloadAttributes) {
-    return SafeFuture.COMPLETE;
+    return SafeFuture.completedFuture(
+        new ForkChoiceUpdatedResult(
+            ForkChoiceUpdatedStatus.SUCCESS,
+            payloadAttributes.map(
+                payloadAttributes1 -> {
+                  Bytes8 payloadId =
+                      Bytes8.leftPad(Bytes.ofUnsignedInt(payloadIdCounter.incrementAndGet()));
+                  payloadIdToHeadAndAttrsCache.invalidateWithNewValue(
+                      payloadId,
+                      new HeadAndAttributes(
+                          forkChoiceState.getHeadBlockHash(), payloadAttributes1));
+                  return payloadId;
+                })));
   }
 
   @Override
-  public SafeFuture<ExecutionPayload> getPayload(final Bytes8 payloadId) {
-    return SafeFuture.failedFuture(new UnsupportedOperationException("getPayload not supported"));
+  public SafeFuture<ExecutionPayload> getPayload(final Bytes8 payloadId, final UInt64 slot) {
+    if (executionPayloadSchema.isEmpty()) {
+      return SafeFuture.failedFuture(
+          new UnsupportedOperationException("getPayload not supported for non-Merge milestones"));
+    }
```

Comment:
Shouldn't we now use `slot` here to get the right schema from Spec?

Reply:
right, fixed